src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -434,7 +434,7 @@ public async Task AnnounceBlocksAsync(List<ChainedHeader> blocksToAnnounce)
                         this.lastSentHeader = bestIndex;
                         this.consensusManagerBehavior.UpdateBestSentHeader(this.lastSentHeader);
 
-                        await peer.SendMessageAsync(new HeadersPayload(headers.ToArray())).ConfigureAwait(false);
+                        await peer.SendMessageAsync(BuildAnnouncedHeaderPayload(this.chainState.BlockStoreTip.Height, headers.ToArray())).ConfigureAwait(false);
```

Comment:
Here you probably want to pass the consensus tip not store tip (but I am not entirely sure)

Reply:
Thats because store tip may take up to 37 seconds to updated (thats the persist batch interval)

Reply:
Mental note we should probably set Chain.StoreTip to be the tip in memory not in disk @fassadlr what do you think?